android/src/main/java/com/ioreactnativecieid/IoReactNativeCieidModule.kt

```diff
@@ -28,10 +29,47 @@ class IoReactNativeCieidModule(reactContext: ReactApplicationContext) :
 
   override fun getName() = NAME
 
+  private fun isSignatureValid(packageName: String, signature: String): Boolean {
+    val pm = reactApplicationContext.packageManager
+    val packageInfo = pm.getPackageInfoCompat(packageName)
+    val signatures = packageInfo.getSignaturesCompat()
+    val sha256List = signatures.map { it.toSha256() }
+    // Check if the given signature is in the SHA-256 list
+    return sha256List.contains(signature)
+  }
+  // Extension function to handle API compatibility for getPackageInfo
+  private fun PackageManager.getPackageInfoCompat(packageName: String): PackageInfo {
+    return if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+      getPackageInfo(packageName, PackageManager.GET_SIGNING_CERTIFICATES)
+    } else {
+      @Suppress("DEPRECATION")
+      getPackageInfo(packageName, PackageManager.GET_SIGNATURES)
+    }
+  }
+  // Extension function to handle API compatibility for getting signatures
+  private fun PackageInfo.getSignaturesCompat(): Array<Signature> {
+    return if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+      signingInfo.apkContentsSigners
+    } else {
+      @Suppress("DEPRECATION")
+      signatures
+    }
+  }
+  // Extension function to convert a signature to its SHA-256 hash
+  private fun Signature.toSha256(): String {
+    val digest = MessageDigest.getInstance("SHA-256")
+    return digest.digest(toByteArray())
+      .joinToString(":") { byte -> "%02x".format(byte).uppercase() }
+  }
+
   @ReactMethod(isBlockingSynchronousMethod = true)
-  fun isAppInstalled(packageName: String): Boolean = try {
+  fun isAppInstalled(packageName: String, signature: String? = null): Boolean = try {
     reactApplicationContext.packageManager.getPackageInfo(packageName, 0)
-    true
+    if(signature == null) {
```

Comment:
```suggestion
    if (signature == null) {
```

I see some formatting issues. Try to format the whole file with Android Studio. There are a few ways to do that, personally I use: `Command+Option+L`.

Reply:
Thanks for the shortcut! Addressed [here](https://github.com/pagopa/io-react-native-cieid/pull/10/commits/eac019a2570d2c99cde7bc9edeadb31290751bef)